### PR TITLE
Fix bug that contract method is duplicate with names as ones from Object.java

### PR DESCRIPTION
**Pull request by leejh903**
### What does this PR do?
This PR solved the issue(#896) 

### Where should the reviewer start?
just SolidityFunctionWrapper.java

### Why is it needed?
As you can see issue(#896), there is compile error when contract method name is duplicate with names as ones from Object.java, especially there are no parameters. This PR adds more condition which append '_' in front of method name in this situation.

### CODE REVIEW

File: codegen/src/main/java/org/web3j/codegen/SolidityFunctionWrapper.java [hunk 1]
```diff
@@ -98,6 +99,8 @@
     private static final Pattern pattern = Pattern.compile(regex);
     private final GenerationReporter reporter;
 
+    private static Set<String> parentsMethodNames = getParentsMethodNames();
```
Comment (iikirilov) [inline codegen/src/main/java/org/web3j/codegen/SolidityFunctionWrapper.java]:
```suggestion
    private static final Set<String> restrictedMethodNames = new HashSet<>(Arrays.asList(Object.class.getDeclaredMethods()));
```

We only need to build this set once

File: codegen/src/main/java/org/web3j/codegen/SolidityFunctionWrapper.java [hunk 2]
```diff
@@ -686,7 +689,7 @@ MethodSpec buildFunction(
 
         // If the solidity function name is a reserved word
         // in the current java version prepend it with "_"
-        if (!SourceVersion.isName(functionName)) {
+        if (!SourceVersion.isName(functionName) || duplicateWithParentsMethods(functionName)) {
```
Comment (iikirilov) [inline codegen/src/main/java/org/web3j/codegen/SolidityFunctionWrapper.java]:
```suggestion
        if (!SourceVersion.isName(functionName) || restrictedMethodNames.contains(functionName)) {
```

File: codegen/src/main/java/org/web3j/codegen/SolidityFunctionWrapper.java [hunk 3]
```diff
@@ -708,6 +711,19 @@ MethodSpec buildFunction(
         return methodBuilder.build();
     }
 
+    private boolean duplicateWithParentsMethods(String functionName) {
```
Comment (iikirilov) [inline codegen/src/main/java/org/web3j/codegen/SolidityFunctionWrapper.java]: these two methods can be removed

File: codegen/src/main/java/org/web3j/codegen/SolidityFunctionWrapper.java (hunk 1 above)
Comment (leejh903) [reply codegen/src/main/java/org/web3j/codegen/SolidityFunctionWrapper.java]: I think suggested code is more better than previous one!

File: codegen/src/main/java/org/web3j/codegen/SolidityFunctionWrapper.java (hunk 3 above)
Comment (leejh903) [reply codegen/src/main/java/org/web3j/codegen/SolidityFunctionWrapper.java]: I removed two methods. Thanks

**Comment by leejh903** [comment]
@iikirilov Thanks for your review. The suggested code has a compile error because the return type of `new HashSet<>(Arrays.asList(Object.class.getDeclaredMethods()));` is HashSet<Method> and it is not HashSet<String>. So I changed your suggested code and add test code related with this code.

**Comment by leejh903** [comment]
@iikirilov I recovered some conflicts

File: codegen/src/test/java/org/web3j/codegen/SolidityFunctionWrapperTest.java [hunk 4]
```diff
@@ -12,11 +12,10 @@
  */
 package org.web3j.codegen;
 
+import java.lang.reflect.Method;
 import java.math.BigInteger;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.List;
+import java.util.*;
```
Comment (iikirilov) [inline codegen/src/test/java/org/web3j/codegen/SolidityFunctionWrapperTest.java]: please remove wildcard

Comment (leejh903) [reply codegen/src/test/java/org/web3j/codegen/SolidityFunctionWrapperTest.java]: I reflected your feedback!

**Comment by leejh903** [comment]
Close this PR because function name collision solved adding prefix in front of function name.

**Pull request closed without merging**